File: rb109/Practice/Easy 1/how_many.rb
# how_many.rb

vehicles = %w(car car truck car SUV truck motorcycle motorcycle car truck)

def count_occurrences(list)
  hsh = Hash.new(0)

  list.each do |elem|
    hsh[elem] += 1
  end

  hsh.each do |element, occurences|
    puts "#{element} => #{occurences}"
  end
end

count_occurrences(vehicles)

File: js210/js210_small_problems/easy_4/how_many.js
"use strict"

function countOccurrences(array) {
  let counts = {};

  for (let i = 0; i < array.length; i += 1) {
    if (counts[array[i]]) {
      counts[array[i]] += 1;
    } else {
      counts[array[i]] = 1;
    }
  }

  let elements = Object.keys(counts);

  elements.forEach(element => console.log(element + ' => ' + counts[element]));
}

const vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck',
                'motorcycle', 'motorcycle', 'car', 'truck'];

countOccurrences(vehicles);
